test/e2e/idle_connection_test.go

```diff
@@ -0,0 +1,634 @@
+			},
+			expectedResponse: webService2,
+		},
+		{
+			description: "Ensure subsequent responses are served by web-service-2",
+			fetchResponse: func(httpClient *http.Client, elbAddress string, cfg *testConfig) (string, string, error) {
+				return idleConnectionFetchResponse(t, httpClient, elbAddress, cfg.routeHost)
+			},
+			expectedResponse: webService2,
+		},
+	}
+
+	deferredActions := []testAction{
+		{
+			description: "Verify the initial response is correctly served by web-service-1",
+			fetchResponse: func(httpClient *http.Client, elbAddress string, cfg *testConfig) (string, string, error) {
+				return idleConnectionFetchResponse(t, httpClient, elbAddress, cfg.routeHost)
+			},
+			expectedResponse: webService1,
+		},
+		{
+			description: "Switch route to web-service-2 and validate Deferred policy allows one final response to be served by web-service-1",
+			fetchResponse: func(httpClient *http.Client, elbAddress string, cfg *testConfig) (string, string, error) {
+				if err := idleConnectionSwitchRouteService(t, cfg.routeName, cfg.ingressController.Name, webService2); err != nil {
+					return "", "", err
+				}
+				return idleConnectionFetchResponse(t, httpClient, elbAddress, cfg.routeHost)
+			},
+			expectedResponse: webService1,
+		},
+		{
+			description: "Ensure subsequent responses are now served by web-service-2",
+			fetchResponse: func(httpClient *http.Client, elbAddress string, cfg *testConfig) (string, string, error) {
+				return idleConnectionFetchResponse(t, httpClient, elbAddress, cfg.routeHost)
+			},
+			expectedResponse: webService2,
+		},
+	}
+
+	for _, tc := range []struct {
+		policy  operatorv1.IngressControllerConnectionTerminationPolicy
+		actions []testAction
+	}{
+		{
+			policy:  operatorv1.IngressControllerConnectionTerminationPolicyImmediate,
+			actions: immediateActions,
+		},
+		{
+			policy:  operatorv1.IngressControllerConnectionTerminationPolicyDeferred,
+			actions: deferredActions,
+		},
+	} {
+		tc := tc
+		t.Run(string(tc.policy), func(t *testing.T) {
+			t.Parallel()
+
+			testName := names.SimpleNameGenerator.GenerateName(strings.ToLower(fmt.Sprintf("idle-connection-close-%s-", tc.policy)))
+			testNs := createNamespace(t, testName)
+			testNs.Labels = map[string]string{testNs.Name + "-shard": testName}
+
+			if err := retry.OnError(retry.DefaultBackoff, func(err error) bool {
+				return apierrors.IsConflict(err)
+			}, func() error {
+				if err := kclient.Get(context.TODO(), client.ObjectKeyFromObject(testNs), testNs); err != nil {
+					return fmt.Errorf("error fetching latest version of namespace %s: %w", testNs.Name, err)
+				}
+				if err := kclient.Update(context.TODO(), testNs); err != nil {
+					t.Logf("conflict updating namespace %s with labels %s: %v, retrying...", testName, testNs.Labels, err)
+					return err
+				}
+				return nil
+			}); err != nil {
+				t.Fatalf("error updating namespace %s with labels %s: %v", testName, testNs.Labels, err)
+			}
```

Comment:
Should we sacrifice the simplicity of the code (and the test execution time too) for a slightly more complex backport merging? 

```suggestion
			testNs := createNamespaceWithLabels(t, testName, map[string]string{testNs.Name + "-shard": testName})
```

```go
func createNamespace(t *testing.T, name string) {
    return createNamespaceWithLabels(t, name, nil)
}
```

Reply:
Is your suggestion to modify the existing `createNamespace` function? I don't want to make the backports painful. And in a test run that runs three hours I fail to see what a Get/Update will add to the mix.

Reply:
> for a slightly more complex backport merging?

For me the question answers itself; why go looking for work/